src/libs/actions/Report.js

```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
         });
 }
 
+/**
+ * Get a simplified version of an IOU report
+ *
+ * @param {Number} reportID
+ * @param {Object} reportData
+ * @param {Number} reportData.transactionID
+ * @param {Number} reportData.amount
+ * @param {String} reportData.currency
+ * @param {String} reportData.created
+ * @param {String} reportData.comment
+ * @param {Object[]} reportData.transactionList
+ * @param {String} reportData.ownerEmail
+ * @param {String} reportData.managerEmail
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportID, reportData) {
+    const transactions = _.map(reportData.transactionList, transaction => ({
+        transactionID: transaction.transactionID,
+        amount: transaction.amount,
+        currency: transaction.currency,
+        created: transaction.created,
+        comment: transaction.comment,
+    }));
+
+    return {
+        reportID,
+        ownerEmail: reportData.ownerEmail,
+        managerEmail: reportData.managerEmail,
+        currency: reportData.currency,
+        transactions,
+    };
+}
+
+/**
+ * Fetches the updated data for an IOU Report and updates the IOU collection in ONYX
+ *
+ * @param {Number} reportID
+ * @param {Object[]} reportHistory
+ */
+function updateIOUReportData(reportID, reportHistory) {
+    const containsIOUAction = _.any(reportHistory, action => action.actionName === 'IOU');
+
+    // If there aren't any IOU actions, we don't need to fetch any additional data
+    if (!containsIOUAction) {
+        return;
+    }
+
+    const otherParticipants = _.chain(reportHistory)
+        .pluck('actorEmail')
+        .unique()
+        .without(currentUserEmail)
+        .value();
+
+    // If we have more than one participant, this is not an IOU
+    if (otherParticipants.length > 1) {
+        return;
+    }
+
+    // If we have an IOU action, get the IOU reportID
+    let iouReportID = 0;
```

Comment:
Yeah, I'm really confused here. If `reportID` is being passed to this method, why are we looking up a different report ID and why does that report have different data than the original report?

Reply:
The `reportID` param is the chat report ID, IOUTransactions are stored as expenses on a separate IOU Report, allowing us to close and switch ownership without tying it too closely to the ChatReport. This has always been the plan.

Reply:
OK, thanks for explaining that. I think it's good context to have, but definitely not apparent from just looking at the code so it will probably confuse others as well. Typically, the best way to combat that is with better code comments that give a bit more context. Would you mind adding a few more comments here to help?

Reply:
I've updated the comment there, let me know if that still doesn't make sense!